sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -727,9 +729,8 @@ class RapidsBufferCatalog(
   def numBuffers: Int = bufferMap.size()
 
   override def close(): Unit = {
-    bufferIdToHandles.values.forEach { handles =>
-      handles.foreach(_.close())
-    }
+    bufferIdToHandles.values.asScala.toSeq.flatMap(_.seq).safeClose()
```

Comment:
The existing code already creates a throwaway collection via `values`. This PR adds more conversions doing it on top of it.

```suggestion
    bufferIdToHandles.forEach((_, bufSeq) => bufSeq.safeClose())
```

Reply:
The thing is that safeClose() can throw. I think iterating in forEach will not complete the iteration if one of safeClose() calls throws.

Reply:
Also, `ConcurrentHashMap.values()` creates a view of it. It shouldn't be expensive to create.

Reply:
but then we have 
- asScala 
- toSeq
- flatMap(_.seq)

Reply:
> The thing is that safeClose()  can throw.

good point

Reply:
> but then we have
> 
> * asScala
> * toSeq
> * flatMap(_.seq)

Hmm, does this make a copy of the values collection? If so, I could use the Java Stream API with a try-catch clause.

Reply:
I thought it would make a scala `Iterable` per https://docs.scala-lang.org/overviews/collections-2.13/conversions-between-java-and-scala-collections.html as `values` returns a java Collection.

Reply:
Yes. Stream API will also make a copy. Not worth it, I thought there is a simple solution. If we had more nested use cases we could add another implicit

Reply:
> Stream API will also make a copy.

I would be surprised if it does. My understanding is that they are lazily evaluated and never be materialized unless you tell it to do so. It is just a fancier version of the `Iterator` API.

Reply:
Checking in REPL there are only two materializations in the original 

`values.asScala.toSeq.flatMap(_.seq)`

from toSeq  retrurning a List and  consequently from the following `flatMap`

This  `bufferIdToHandles.values.asScala.flatten.safeClose()` saves toSeq but we need to fix our scala-2.12 implictis.


```scala
scala> val cm = new java.util.concurrent.ConcurrentHashMap[String,Seq[String]]()
val cm: java.util.concurrent.ConcurrentHashMap[String,Seq[String]] = {}

scala> cm.put("test_key", Seq("test_value_1", "test_value_2"))
val res11: Seq[String] = null

scala> cm.values
val res12: java.util.Collection[Seq[String]] = [List(test_value_1, test_value_2)]

scala> cm.values.asScala
warning: 1 deprecation (since 2.13.0); for details, enable `:setting -deprecation` or `:replay -deprecation`
val res13: Iterable[Seq[String]] = Iterable(List(test_value_1, test_value_2))

scala> cm.values.asScala.toSeq
warning: 1 deprecation (since 2.13.0); for details, enable `:setting -deprecation` or `:replay -deprecation`
val res14: Seq[Seq[String]] = List(List(test_value_1, test_value_2))

scala> cm.values.asScala.toSeq.flatMap(_.seq)
warning: 2 deprecations (since 2.13.0); for details, enable `:setting -deprecation` or `:replay -deprecation`
val res15: Seq[String] = List(test_value_1, test_value_2)
```

Reply:
Surprising that `flatMap` materializes.. 🙁 Good to know. Yeah perhaps it's a good idea to add a new `safeClose()` version for nested collections if we have enough use cases. It could use `bufferIdToHandles.values.asScala.flatten.safeClose()` you suggested.

Reply:
> Surprising that flatMap materializes..

The culprit is toSeq, which you only have to do because Scala 2.12 version of implicitcs is written for SeqLike. You can update it in this PR to use Iterable as in https://github.com/NVIDIA/spark-rapids/blob/8ef2e41d4b4126eecf9130043a2b7bedf2f4263f/sql-plugin/src/main/scala-2.13/com/nvidia/spark/rapids/implicits.scala#L86C2-L112C4

Reply:
Thanks for the suggestion. I think it would be better to be done in a separate PR though, created https://github.com/NVIDIA/spark-rapids/issues/11314.